quinn/src/endpoint.rs

```diff
@@ -199,6 +201,11 @@ where
                 break;
             }
         }
+        if !endpoint.incoming.is_empty() && endpoint.incoming_live {
```

Comment:
One more thought: Why is `incoming_live` important here? If it doesn't exist anymore, wouldn't it be also fair to get rid of this `Waker` (by at least doing `take`) in order to remove any potential cross-references?

If deregistering `incoming` would already remove the waker, then this check would also be not important.

Reply:
Good catch. I think this was left over from a long time ago. It's definitely not doing anything useful now; removed.

Reply:
I've seen https://github.com/quinn-rs/quinn/pull/883/commits/d215e586d5c57370599bcb7076cc150640091fc6

Wouldn't it still be to helpful whether to decide to store something in `incoming` at all?
Without it, that list of pending incoming connections grows indefinitely? Not sure if there is any bound to it.